packages/medusa/src/api-v2/auth/[scope]/[authProvider]/route.ts

```diff
@@ -23,16 +23,18 @@ export const GET = async (req: MedusaRequest, res: MedusaResponse) => {
   const authResult = await service.authenticate(authProvider, authData)
 
   const { success, error, authUser, location } = authResult
+
   if (location) {
     res.redirect(location)
     return
   }
 
   if (success) {
     const { jwt_secret } = req.scope.resolve("configModule").projectConfig
-
     const token = jwt.sign(authUser, jwt_secret)
 
+    req.session.auth_user = authUser
```

Comment:
Doesn't this happen in the middleware (and if not, maybe it should happen there)?

Reply:
This was added to make it work for the admin as the admin worked with sessions. The authenticate middleware doesn't "create" sessions. We probably need a create session endpoint or a config to enable that in the authenticate endpoint at some point or rework the admin to use token auth.

I'll remove this from here for now.